File: Math/leetcode_07.py

# Problem Link ====>> https://leetcode.com/problems/reverse-integer/description/

import sys


class Solution:
    def reverse(self, x: int) -> int:
        reversed_number = 0
        while x != 0:
            remainder = x % 10
            reversed_number = reversed_number * 10 + remainder
            x //= 10

        if reversed_number > sys.maxint or reversed_number < -sys.maxint - 1:
            return 0
        else:
            return reversed_number


if __name__ == "__main__":
    x = -12345
    solution = Solution()
    reversed_integer = solution.reverse(x)

    if reversed_integer == 0:
        print("The reversed integer overflows the 32-bit signed integer range.")
    else:
        print(f"The reversed integer of {x} is: {reversed_integer}")

File: Math/leetcode_07.cpp

// Problem Link ====>> https://leetcode.com/problems/reverse-integer/description/

#include <iostream>
#include <climits>
using namespace std;

class Solution
{
public:
    int reverse(int x)
    {
        int rev = 0;
        while (x != 0)
        {
            int rem = x % 10;
            rev = rev * 10 + rem;
            x /= 10;
        }

        // Check if the reversed integer overflows the 32-bit signed integer range.
        if (rev > INT_MAX || rev < INT_MIN)
        {
            return 0;
        }
        else
        {
            return rev;
        }
    }
};

int main()
{
    int x = -12345;
    Solution solution;

    int reversed = solution.reverse(x);

    if (reversed == 0)
    {
        cout << "The reversed integer overflows the 32-bit signed integer range." << endl;
    }
    else
    {
        cout << "The reversed integer of " << x << " is: " << reversed << endl;
    }

    return 0;
}